src/Platforms/SQLitePlatform.php

```diff
@@ -765,45 +765,46 @@ private function getSimpleAlterTableSQL(TableDiff $diff): array|false
         return $sql;
     }
 
-    /** @return string[] */
-    private function getColumnNamesInAlteredTable(TableDiff $diff, Table $oldTable): array
+    /**
+     * Based on the table diff, returns a map where the keys are the lower-case old column names and the values are the
+     * new column names. If the column was dropped, it is not present in the map.
+     *
+     * @return array<string, string>
+     */
+    private function getDiffColumnNameMap(TableDiff $diff): array
     {
-        $columns = [];
+        $oldTable = $diff->getOldTable();
+
+        $map = [];
 
         foreach ($oldTable->getColumns() as $column) {
-            $columnName                       = $column->getName();
-            $columns[strtolower($columnName)] = $columnName;
+            $columnName                   = $column->getName();
+            $map[strtolower($columnName)] = $columnName;
         }
 
         foreach ($diff->getDroppedColumns() as $column) {
-            $columnName = strtolower($column->getName());
-            if (! isset($columns[$columnName])) {
-                continue;
-            }
-
-            unset($columns[$columnName]);
+            unset($map[strtolower($column->getName())]);
         }
 
         foreach ($diff->getChangedColumns() as $columnDiff) {
-            $oldColumnName                       = $columnDiff->getOldColumn()->getName();
-            $newColumnName                       = $columnDiff->getNewColumn()->getName();
-            $columns[strtolower($oldColumnName)] = $newColumnName;
-            $columns[strtolower($newColumnName)] = $newColumnName;
+            $columnName                   = $columnDiff->getOldColumn()->getName();
+            $map[strtolower($columnName)] = $columnDiff->getNewColumn()->getName();
         }
 
         foreach ($diff->getAddedColumns() as $column) {
-            $columnName                       = $column->getName();
-            $columns[strtolower($columnName)] = $columnName;
+            $columnName                   = $column->getName();
+            $map[strtolower($columnName)] = $columnName;
         }
 
-        return $columns;
+        return $map;
     }
 
-    /** @return Index[] */
-    private function getIndexesInAlteredTable(TableDiff $diff, Table $oldTable): array
+    /** @return array<Index> */
```

Comment:
Here and in one more method, the resulting array contains a mix of numeric and string keys. This likely causes some edge-case bugs, but I'm not going to change this here.